packages/http-server-csharp/eng/scripts/emit-scenarios.js

```diff
@@ -0,0 +1,278 @@
+/* eslint-disable no-console */
+import { execa } from "execa";
+import { copy, pathExists } from "fs-extra";
+import { mkdir, readFile, rm, writeFile } from "fs/promises";
+import { globby } from "globby";
+import inquirer from "inquirer";
+import ora from "ora";
+import pLimit from "p-limit";
+import { basename, dirname, join, resolve } from "path";
+import pc from "picocolors";
+import { fileURLToPath } from "url";
+import yargs from "yargs";
+import { hideBin } from "yargs/helpers";
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+const tspConfig = join(__dirname, "tspconfig.yaml");
+
+const basePath = join(__dirname, "../..");
+const logDirRoot = join(basePath, "temp", "emit-scenarios-logs");
+const reportFilePath = join(logDirRoot, "report.txt");
+const testScenarioPath = join(basePath, "test", "scenarios");
+const ignoreFilePath = join(testScenarioPath, ".testignore");
+
+const argv = yargs(hideBin(process.argv))
+  .option("input", {
+    alias: "i",
+    type: "string",
+    description: "Input spec folder",
+    default: join(basePath, "..", "http-specs", "specs"),
+  })
+  .option("build", {
+    type: "boolean",
+    describe: "Build the generated projects",
+    default: true,
+  })
+  .option("interactive", {
+    type: "boolean",
+    describe: "Enable interactive mode",
+    default: false,
+  })
+  .help().argv;
+
+const specDir = resolve(argv.input);
+
+// Initialize the port number
+let portNumber = 65000;
+
+// Remove the log directory if it exists.
+async function clearDirectory(dirRoot) {
+  if (await pathExists(dirRoot)) {
+    await rm(dirRoot, { recursive: true, force: true });
+  }
+}
+
+async function copySelectiveFiles(extension, sourceDir, targetDir) {
```

Comment:
```suggestion
async function copySelectedFiles(extension, sourceDir, targetDir) {
```

Reply:
Here I do mean selective files as the first param is essentially a match pattern